src/common/components/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,66 @@
+import { FC } from "react";
+
+import { useSelect } from "downshift";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faAngleUp, faAngleDown } from "@fortawesome/free-solid-svg-icons";
+import {
+  DropdownContainer,
+  DropdownHeader,
+  DropdownList,
+  DropdownListItem,
+  UpArrow,
+  DownArrow,
+} from "./Dropdown.styled";
+
+interface OptionType {
+  value: string;
+  label: string;
+}
+
+export interface Props {
+  items: OptionType[];
+  // onChange: (selectedItem: string) => void;
+}
+
+const Dropdown: FC<Props> = ({ items }) => {
+  const {
+    isOpen,
+    selectedItem,
+    getToggleButtonProps,
+    getMenuProps,
+    highlightedIndex,
+    getItemProps,
+  } = useSelect({ items });
+
+  return (
+    <DropdownContainer>
+      <DropdownHeader {...getToggleButtonProps()} isOpen={isOpen}>
+        {(selectedItem && selectedItem.label) || "---"}
+        {isOpen ? (
+          <UpArrow>
+            <FontAwesomeIcon icon={faAngleUp} />
+          </UpArrow>
+        ) : (
+          <DownArrow>
+            <FontAwesomeIcon icon={faAngleDown} />
+          </DownArrow>
+        )}
+      </DropdownHeader>
+      <DropdownList {...getMenuProps()} isOpen={isOpen}>
+        {isOpen &&
+          items.map((item, index) => (
+            <DropdownListItem
+              isHighlighted={highlightedIndex === index}
+              key={`${item.value}${index}`}
```

Comment:
im not sure if this is a good key since an item could change position in list, and then be considered a different item. is that ok?  theres also an edge case that this concatenation can lead to an ambiguous key.  wdyt?

Reply:
We could do item.value alone? As the value of a dropdown should be unique.

Reply:
i think so